### PR TITLE
54213 View Subjects - Add New Subject from page

### PR DESCRIPTION
**Pull request by naid**
54213 View Subjects - Add New Subject from page

### CODE REVIEW

File: app/Http/Controllers/Supervisor/SubjectController.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            'subjects' => $subjects,
+        ]);
+    }
+
+    public function store(CreateSubjectRequest $request)
+    {
+        try {
+            $subjectData = $request->all();
+            $subject = new Subject;
+            $subject->create($subjectData);
```
Comment (wataridori) [inline app/Http/Controllers/Supervisor/SubjectController.php]: You can use repository for `create` action. Do not create an instance and use it to call `Eloquent` method.

Comment (naid) [inline app/Http/Controllers/Supervisor/SubjectController.php]: This is now fixed. Thanks.

File: app/Http/Controllers/Supervisor/SubjectController.php [hunk 2]
```diff
@@ -18,6 +19,28 @@ public function index()
...
+
+            $this->subjectRepository->create($subjectData);
+            session()->flash('flash_message', trans('message.success_subject'));
+        } catch (Exception $e) {
+            session()->flash('flash_error', trans('message.error_subject'));
+        }
+
+        $subjects = $this->subjectRepository->getRowsPaginated();
+
+        return view('supervisor/subjects/list', [
```
Comment (dinhhoanglong91) [inline app/Http/Controllers/Supervisor/SubjectController.php]: Sorry I forgot commenting the previous PR. 
List of subjects should be shown in `supervisor/subjects/index`

File: resources/views/supervisor/subjects/list.blade.php [hunk 3]
```diff
@@ -8,6 +8,22 @@
                     <div class="panel-heading">
                         {{ trans('message.list_of_subjects') }}
                     </div>
+            <div class="panel-body">
```
Comment (dinhhoanglong91) [inline resources/views/supervisor/subjects/list.blade.php]: indent

**Comment by dinhhoanglong91** [comment]
ok

**Comment by wataridori** [comment]
OK

File: resources/views/supervisor/subjects/index.blade.php [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+@section('content')
+    <div class="container">
+        <div class="row">
+            <div class="col-md-12">
+                <div class="panel panel-default">
+                    <div class="panel-heading">
+                        {{ trans('message.list_of_subjects') }}
+                    </div>
+                    <div class="panel-body">
+                        {{ Form::open(['route' => ['supervisor.subjects.store'], 'files' => false]) }}
```
Comment (nguyenthanhtung88) [inline resources/views/supervisor/subjects/index.blade.php]: Default value for key `'files'` is `false` right? you should remove this.

**Pull request merged**